Forms/MainForm.cs

```diff
@@ -75,12 +78,22 @@ private void UpdateConfigurationSubstitutions()
         {
             Configuration.DefaultSubstitutions.Clear();
             Configuration.DefaultSubstitutions.AddRange(substitutionList1.Substitutions);
+            LoadConfigurationDetails();
         }
 
 
         private void LoadConfigurationDetails()
         {
-            txtWordList.Text = Configuration.DefaultWords.ToString();
+            string wordtext = Configuration.DefaultWords.ToString();
+            if (txtWordList.Text != null && txtWordList.Text != wordtext && txtWordList.Text != wordtext + " ")

```

Comment:
In terms of user experience I think the best approach for the word list
would be timer-based, ignoring focus. I think that's going to involve some
multi-thread protection to prevent the timer updating the gui at the same
time as some other event. That will prevent updating samples with
half-words, or while the user is busy typing, and will also allow
copy-paste.

But that will still leave the problem of erasing trailing spaces. That's
caused by my removing empty items when parsing the list into words. I think
that's a good thing but it means the conversion back to text for the box
will never include trailing spaces.

Maybe the best way around that would be to set a flag of some sort (maybe
an extra argument with a default value) that the text box does not need
updating. Actually the only time it should need ever need updating via code
is during the initial load, come to think of it.
On Nov 16, 2015 8:35 AM, "Pascal" <notifications@github.com> wrote:

> In Forms/MainForm.cs
> <https://github.com/fireout/keepasssequencer/pull/17#discussion_r44929283>
> :
>
> >          }
> >
> >
> >          private void LoadConfigurationDetails()
> >          {
> > -            txtWordList.Text = Configuration.DefaultWords.ToString();
> > +            string wordtext = Configuration.DefaultWords.ToString();
> > +            if (txtWordList.Text != null && txtWordList.Text != wordtext && txtWordList.Text != wordtext + " ")
>
> I still had an issue with the PR version where when I would add a second
> space, the field would reload and reposition the cursor...
>
> You are right, basing the operation only on the focus might skip the
> initial load, we can probably add a check with the input's text length...
>
> If you feel an other approach might not be as straight forward as I
> expect, I can approve this PR and work on this later on... At least for
> now, the UI experience is a lot better than what it was before.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fireout/keepasssequencer/pull/17/files#r44929283>.
>